github-pullrequest-plugin/src/test/java/org/jenkinsci/plugins/github/pullrequest/trigger/check/PullRequestToCauseConverterTest.java

```diff
@@ -116,4 +126,32 @@ public void shouldReturnCauseOnWhen1OpenedWeGetSecondOneAndHaveEvents() throws E
         assertThat("reason in cause", cause.getReason(), containsString("open"));
         assertThat("pr num in cause", cause.getNumber(), is(2));
     }
+
+    /**
+     * Test trigger configuration of:
+     *
+     *     1.) Skip PR if label is not present (when label is not present)
+     *     2.) Cause PR if commit changed (when commit has changed)
+     *
+     * Expected result is that the PR should be skipped. No causes should be
+     * identified.
+     */
+    @Test
+    public void shouldSkipEvenThoughNewCommit() throws Exception {
+        GHCommitPointer commitPtr = mock(GHCommitPointer.class);
+        when(local.getPulls()).thenReturn(ImmutableMap.of(3, localPR));
+        when(localPR.getHeadSha()).thenReturn("this is not the sha you are looking for");
+        when(remotePR.getNumber()).thenReturn(3);
+        when(remotePR.getState()).thenReturn(GHIssueState.OPEN);
+        when(remotePR.getHead()).thenReturn(commitPtr);
+        when(trigger.getEvents()).thenReturn(asList(
+                new GitHubPRLabelNotExistsEvent(new GitHubPRLabel("notfound"), true),
+                new GitHubPRCommitEvent()));
+
+        List<GitHubPRCause> causes = from(ImmutableSet.of(remotePR))
+                .transform(toGitHubPRCause(local, tlRule.getListener(), trigger))
+                .filter(notNull()).toList();
+
+        assertThat("no cause", causes, hasSize(0));
```

Comment:
This test code was more or less copied from from [GitHubPRTrigger (linked)](https://github.com/jenkinsci/github-integration-plugin/blob/master/github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/GitHubPRTrigger.java#L371-L378) where it appears to apply all the trigger causes to the PR. I can't say I am able to follow the logic in there very clearly but I don't see where it would bail with a `skip=true` cause. I also don't see where it would stop evaluating causes when it found a matching `skip=true` cause - as the documentation seems to indicate that it should. 

I'm happy to try to make this fix but I might need a little more guidance on what the code is supposed to be doing here.